filters/auth/jwt_validation.go

```diff
@@ -0,0 +1,143 @@
+package auth
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/MicahParks/keyfunc"
+	"github.com/golang-jwt/jwt"
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+)
+
+const (
+	JwtValidationName = "jwtValidation"
+)
+
+type (
+	jwtValidationSpec struct {
+		options TokenintrospectionOptions
+	}
+
+	jwtValidationFilter struct {
+		jwksUri string
+	}
+)
+
+var m sync.Mutex
+
+//the map of jwks keyfunctions stored per jwksUri
+var jwksMap map[string]*keyfunc.JWKs = make(map[string]*keyfunc.JWKs)
+
+func NewJwtValidationWithOptions(o TokenintrospectionOptions) filters.Spec {
+	return &jwtValidationSpec{
+		options: o,
+	}
+}
+
+func (s *jwtValidationSpec) Name() string {
+	return JwtValidationName
+}
+
+func (s *jwtValidationSpec) CreateFilter(args []interface{}) (filters.Filter, error) {
+	sargs, err := getStrings(args)
+	if err != nil {
+		return nil, err
+	}
+	if len(sargs) != 1 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	issuerURL := sargs[0]
+
+	cfg, err := getOpenIDConfig(issuerURL)
+	if err != nil {
+		return nil, err
+	}
+
+	f := &jwtValidationFilter{
+		jwksUri: cfg.JwksURI,
+	}
+
+	return f, nil
+}
+
+func getKeyFunction(url string) (jwks *keyfunc.JWKs, err error) {
+	m.Lock()
+	defer m.Unlock()
+
+	if jwks, ok := jwksMap[url]; ok {
+		return jwks, nil
+	}
+
+	refreshInterval := time.Hour
+	refreshRateLimit := time.Minute * 5
+	refreshTimeout := time.Second * 10
+	refreshUnknownKID := true
```

Comment:
all these vars can be package global const instead to omit creating garbage

Reply:
@szuecs keyfunc.Options{} requires pointers not just values. That's why I have not declared all this settings as constants.

Reply:
Interesting you can't get a ptr from a const, I just checked it.

Reply:
You could create a global un-exported variable that would serve the purpose of avoiding allocation/garbage collection.